src/services/arweave/WarpDataProvider.ts

```diff
@@ -1,18 +1,25 @@
 import { Warp, WarpFactory } from 'warp-contracts';
 
-import { ArNSContractState, SmartweaveContractSource } from '../../types';
+import {
+  ArNSContractState,
+  ArweaveTransactionId,
+  SmartweaveContractSource,
+} from '../../types';
 
 export class WarpDataProvider implements SmartweaveContractSource {
   private _warp: Warp;
+  contractId: any;
 
-  constructor() {
+  constructor(contractId: ArweaveTransactionId) {
```

Comment:
this constrains eaach instance to a single contract id, rather than a global WarpProvider that can fetch contract state for any contract id.

Reply:
Is it a bad thing to have a single instance of a class per contract?